internal/k8s/reconciler/deployer.go

```diff
@@ -87,7 +87,26 @@ func (d *GatewayDeployer) ensureServiceAccount(ctx context.Context, config apigw
 		return nil
 	}
 
-	return d.client.EnsureServiceAccount(ctx, gateway, serviceAccount)
+	if err := d.client.EnsureServiceAccount(ctx, gateway, serviceAccount); err != nil {
+		return err
+	}
+
+	role := config.RoleFor(gateway)
+	if role == nil {
+		return nil
+	}
```

Comment:
Following the pattern already established by the `ServiceAccount` code immediately above: return `nil` if we shouldn't be creating a `Role` and move along.